### ISSUE TITLE
Avoid deduplication of objects when forking

### ISSUE THREAD

**Issue by sad75** [go-gitea/gitea#24731]
### Feature Description

At fork time, Gitea creates a Git repository for the new project. This repository is a copy of the original project. When forking large projects, it consumes a lot of disk space. 

Other git servers make use of the [Git alternates feature](https://git-scm.com/docs/gitrepository-layout#Documentation/gitrepository-layout.txt-objectsinfoalternates) :

- Github: Fork would not duplicate (on the server side) the full repository, as explained in [Counting Objects](https://github.blog/2015-09-22-counting-objects/#your-very-own-fork-of-rails)
- GitLab: [How Git object deduplication works in GitLab](https://docs.gitlab.com/ee/development/git_object_deduplication.html)

It would be nice to have such an optimization in Gitea.

**Comment by silverwind** [comment]
GitHub's deduplication method also has the downside that pull requests are actually part of the original repo, so these can be technically be made to appear as part of the original repo on the UI as has happend on GitHub's DMCA repo in the past.

**Comment by lunny** [comment]
Yes, so we need to find object's origin repository if we use this method. I have sent #18459 to try to have a warning once view a commit which actually don't belongs to the main repository.
